src/rust/ide/lib/view/src/graph_editor/component/project_name.rs

```diff
@@ -0,0 +1,314 @@
+//! This module provides a view for project's name which can be used to edit it.
+
+use crate::prelude::*;
+
+use enso_frp as frp;
+use ensogl::data::color;
+use ensogl::display;
+use ensogl::display::Attribute;
+use ensogl::display::Buffer;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::scene::Scene;
+use ensogl::display::shape::text::text_field::TextField;
+use ensogl::display::shape::text::text_field::TextFieldProperties;
+use ensogl::display::shape::*;
+use ensogl::display::Sprite;
+use ensogl::display::world::World;
+use ensogl::gui::component::Animation;
+use ensogl::gui::component;
+use logger::enabled::Logger;
+use logger::AnyLogger;
+use nalgebra::Vector2;
+use ensogl::animation::linear_interpolation;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+const HIGHLIGHTED_TEXT_COLOR : color::Rgba = color::Rgba::new(1.0,1.0,1.0,1.0);
+const DARK_GRAY_TEXT_COLOR   : color::Rgba = color::Rgba::new(1.0,1.0,1.0,0.6);
+
+/// Project name used as a placeholder in ProjectName view.
+pub const UNKNOWN_PROJECT_NAME:&str = "[UNKNOWN NAME]";
+
+
+
+// =============
+// === Shape ===
+// =============
+
+mod shape {
+    use super::*;
+
+    ensogl::define_shape_system! {
+            (style:Style, selection:f32) {
+                let bg_color = color::Rgba::new(0.0,0.0,0.0,0.000_001);
+                Plane().fill(bg_color).into()
+            }
+        }
+}
+
+
+
+// =================
+// === FrpInputs ===
+// =================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct FrpInputs {
+    /// Rename the project.
+    pub rename     : frp::Source<String>,
+    /// Reset the project name to the one before editing.
+    pub reset_name : frp::Source<()>
+}
+
+impl FrpInputs {
+    /// Create new FrpInputs.
+    pub fn new(network:&frp::Network) -> Self {
+        frp::extend! {network
+            def reset_name = source();
+            def rename     = source();
+        }
+        Self{reset_name,rename}
+    }
+}
+
+
+
+// ==================
+// === FrpOutputs ===
+// ==================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct FrpOutputs {
+    /// Emits the new project name when it's renamed.
+    pub renamed : frp::Source<String>
```

Comment:
this should be named "name" . It doesn't matter from frp perspective fit changed because of renaming or other action - this is the updated name of the project.